controller/service.go

```diff
@@ -106,10 +108,31 @@ func (c *controller) convergeBalancer(l log.Logger, key string, svc *v1.Service)
 			c.clearServiceState(key, svc)
 			lbIPs = []net.IP{}
 		}
-		// User set or changed the desired LB IP, nuke the
-		// state. allocateIP will pay attention to LoadBalancerIP and try
-		// to meet the user's demands.
-		if svc.Spec.LoadBalancerIP != "" {
+		// User set or changed the desired LB IP(s), nuke the
+		// state. allocateIP will pay attention to LoadBalancerIP(s) and try
+		// to meet the user's demands. annotationLoadBalancerIPs takes higher
+		// precedence over svc.Spec.LoadBalancerIP.
+		// TODO: remove check for svc.Spec.LoadBalancerIP when this field is removed
+		// from service spec.
+		desiredLbIPs, err := c.getDesiredLbIPs(svc)
```

Comment:
Can you have the LoadBalancerIP logic embedded in `getDesiredLbIPs` ? I think it will remove duplication and in general it will express what `getDesiredIPs` means.
So if the user set the annotation use that, otherwise fallback to Status.LoadBalancerIP .

Reply:
Yes! this is an elegant approach. made the changes now.